src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -1766,6 +1766,78 @@ static void Main()
             CompileAndVerify(source, expected);
         }
 
+        [WorkItem(19609, "https://github.com/dotnet/roslyn/issues/19609")]
+        [Fact]
+        public void SpillArrayAssign2()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+class Program
+{
+    static int[] array = new int[5];
+
+    static void Main(string[] args)
+    {
+        try
+        {
+            System.Console.WriteLine(""test not awaited"");
+            TestNotAwaited().Wait();
+        }
+        catch
+        {
+            System.Console.WriteLine(""exception thrown"");
+        }
+
+    System.Console.WriteLine();
+
+        try
+        {
+            System.Console.WriteLine(""test awaited"");
+            TestAwaited().Wait();
+        }
+        catch
+        {
+            System.Console.WriteLine(""exception thrown"");
+        }
+
+    }
+
+    static async Task TestNotAwaited()
+    {
+        array[6] = Foo1();
+    }
+
+    static async Task TestAwaited()
+    {
+        array[6] = await Foo();
```

Comment:
Polycheck

Reply:
Right. I just copied this from the repro.

---
In reply to: [150088747](https://github.com/dotnet/roslyn/pull/23103#discussion_r150088747) [](ancestors = 150088747)